junit-jupiter-api/src/main/java/org/junit/jupiter/api/DynamicContainer.java

```diff
@@ -84,4 +85,10 @@ private DynamicContainer(String displayName, Stream<? extends DynamicNode> child
 		return children;
 	}
 
+	@Override
+	public DynamicContainer withTestSource(TestSource testSource) {
+		DynamicContainer clone = new DynamicContainer(getDisplayName(), getChildren());
```

Comment:
methinks we should avoid _cloning_ whenever possible.

Similarly, if we have copy-constructor, why not pass in the test source as well? In other words, do we need to make the test source field mutable?